### PR TITLE
[Snyk] Security upgrade transformers from 4.30.2 to 5.0.0rc3

### PR DESCRIPTION
**Pull request by jhamot**
snyk-top-banner

#### Snyk changed the following file(s):

- `requirements.txt`

#### Breaking Change Risk

Merge Risk: High

> **Notice:** This assessment is enhanced by AI.

---

> [!IMPORTANT]
>
> - Check the changes in this PR to ensure they won't cause issues with your project.
> - Max score is 1000. Note that the real score may have changed since the PR was raised.
> - This PR was automatically created by Snyk using the credentials of a real user.
> - Some vulnerabilities couldn't be fully fixed and so Snyk will still find them when the project is tested again. This may be because the vulnerability existed within more than one direct dependency, but not all of the affected dependencies could be upgraded.

---

**Note:** _You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs._

For more information: 
🧐 [View latest project report](https://app.snyk.io/org/lucasfilm/project/097774dc-056b-450a-947b-0ba352e607d9?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)
📜 [Customise PR templates](https://docs.snyk.io/scan-using-snyk/pull-requests/snyk-fix-pull-or-merge-requests/customize-pr-templates?utm_source=github&utm_content=fix-pr-template)
🛠 [Adjust project settings](https://app.snyk.io/org/lucasfilm/project/097774dc-056b-450a-947b-0ba352e607d9?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)
📚 [Read about Snyk's upgrade logic](https://docs.snyk.io/scan-with-snyk/snyk-open-source/manage-vulnerabilities/upgrade-package-versions-to-fix-vulnerabilities?utm_source=github&utm_content=fix-pr-template)

---

**Learn how to fix vulnerabilities with free interactive lessons:**

🦉 [Deserialization of Untrusted Data](https://learn.snyk.io/lesson/insecure-deserialization/?loc&#x3D;fix-pr)

[//]: # 'snyk:metadata:{"breakingChangeRiskLevel":"high","FF_showPullRequestBreakingChanges":false,"FF_showPullRequestBreakingChangesWebSearch":true,"customTemplate":{"variablesUsed":[],"fieldsUsed":[]},"dependencies":[{"name":"transformers","from":"4.30.2","to":"5.0.0rc3"}],"env":"prod","issuesToFix":["SNYK-PYTHON-TRANSFORMERS-15166618"],"prId":"b872d8b2-5168-4ce3-8262-a2b3dae39a40","prPublicId":"b872d8b2-5168-4ce3-8262-a2b3dae39a40","packageManager":"pip","priorityScoreList":[214],"projectPublicId":"097774dc-056b-450a-947b-0ba352e607d9","projectUrl":"https://app.snyk.io/org/lucasfilm/project/097774dc-056b-450a-947b-0ba352e607d9?utm_source=github&utm_medium=referral&page=fix-pr","prType":"fix","templateFieldSources":{"branchName":"default","commitMessage":"default","description":"default","title":"default"},"templateVariants":["updated-fix-title","priorityScore"],"type":"auto","upgrade":[],"vulns":["SNYK-PYTHON-TRANSFORMERS-15166618"],"patch":[],"isBreakingChange":false,"remediationStrategy":"vuln"}'

### CODE REVIEW

**Comment by jhamot** [comment]
Merge Risk: High

The upgrade from `transformers` v4 to v5 is a major release with significant and extensive breaking changes. It requires careful migration as it fundamentally alters the library's architecture.  
  
**Key Breaking Changes:**  
  
*   **PyTorch-Only Backend:** Support for TensorFlow and JAX/Flax has been completely removed. The library now exclusively supports PyTorch. All code using `TFAutoModel`, `FlaxAutoModel`, and other non-PyTorch components must be migrated.  
*   **Quantization API:** The `load_in_4bit` and `load_in_8bit` arguments for model loading have been removed. A new `quantization_config` object must be used instead.  
*   **Tokenization Overhaul:** The distinction between "slow" and "fast" tokenizers has been eliminated in favor of a unified API that automatically selects the best backend.  
*   **Authentication Parameter:** The `use_auth_token` parameter in `from_pretrained` methods is now `token`.  
*   **CLI Changes:** The command-line interface has been revamped. `transformers-cli` is replaced by `transformers`, and the server functionality of `transformers chat` has been moved to a new `transformers serve` command.  
*   **Removed Features:** Several features have been removed, including head masking, head pruning, and relative positional biases in some models.  
*   **HTTP Client:** The internal HTTP client was switched from `requests` to `httpx`, which may require changes to error handling logic for network requests.  
  
Source: [Migration Guide](https://github.com/huggingface/transformers/blob/main/docs/source/en/migration_guide_v5.md), [Release Notes](https://github.com/huggingface/transformers/releases)  
  
**Recommendation:** This upgrade requires significant code refactoring, especially for projects using TensorFlow or JAX. Developers must follow the official migration guide closely to adapt to the new PyTorch-only architecture, updated quantization methods, and other API changes.

> **Notice 🤖:** This content was augmented using **artificial intelligence**. AI-generated content **may** contain **errors** and should be **reviewed** for accuracy before use.
